### ISSUE TITLE
Stack Overflow on Federated Schema Gateway Service in 13.5.0

### ISSUE THREAD

**Issue by albernhagen** [ChilliCream/graphql-platform#6494]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

We have a UI that issues a high number of requests to our gateway service, a number of which do queries in parallel to different domain services.  After some time (sorry I don't have better numbers), we get a stack overflow error:

```
Stack overflow.
Repeat 104670 times:
--------------------------------
at System.HashCode.Combine[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Int32, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon, Int32)
at HotChocolate.Path+RootPathSegment.GetHashCode()
```

### Steps to reproduce

I wish I could share our repos but they are private.  In general, it is:

1. Setup a federated schema service with domain services
2. Issue a high number of client-side concurrent requests to the gateway service of which certain queries include types from both domain services in the same query
3. It'll eventually fail as noted above.

### Relevant log output

```shell
Noted in the bug description.
```

### Additional Context?

This happens when upgrading to 13.5.0 but does not happen with 13.4.0.

### Version

13.5.0

**Comment by michaelstaib** [comment]
Thanks for reporting ... I will have a look at this and think I know why this is happening.

**Comment by michaelstaib** [comment]
Its fixed with [13.6.0-preview.4](https://github.com/ChilliCream/graphql-platform/releases/tag/13.6.0-preview.4)
